src/main/java/org/opentripplanner/graph_builder/module/ned/ElevationModule.java

```diff
@@ -47,24 +62,46 @@
 
     private static final Logger log = LoggerFactory.getLogger(ElevationModule.class);
 
-    private ElevationGridCoverageFactory gridCoverageFactory;
+    private final ElevationGridCoverageFactory gridCoverageFactory;
+    private final boolean readCachedElevations;
+    private final boolean writeCachedElevations;
+    private final File cachedElevationsFile;
+
+    private HashMap<String, PackedCoordinateSequence> cachedElevations;
 
     private Coverage coverage;
 
     // Keep track of the proportion of elevation fetch operations that fail so we can issue warnings.
-    private int nPointsEvaluated = 0;
-    private int nPointsOutsideDEM = 0;
+    private AtomicInteger nPointsEvaluated = new AtomicInteger(0);
+    private AtomicInteger nPointsOutsideDEM = new AtomicInteger(0);
```

Comment:
Add comment describing why AtomicInteger needed (I assume it's because this is now multi-threaded)